File: src/main/webapp/app/shared/model/custom/effectivenessPhaseModels/final-effectiveness-phase-report-model.ts
import { IDetailFinalEffectivenessPhaseReportModel } from './detail-final-effectiveness-phase-report-model';

export interface IFinalEffectivenessPhaseReportModel {
    levelId?: number;
    levelTitle?: string;
    finishedCount?: number;
    averageEffectiveness?: number;
    detailFinalEffectivenessPhaseReportModels?: IDetailFinalEffectivenessPhaseReportModel[];
}

export class FinalEffectivenessPhaseReportModel implements IFinalEffectivenessPhaseReportModel {
    constructor(
        public levelId?: number,
        public levelTitle?: string,
        public finishedCount?: number,
        public averageEffectiveness?: number,
        public detailFinalEffectivenessPhaseReportModels?: IDetailFinalEffectivenessPhaseReportModel[]
    ) {}
}

File: src/main/java/com/marineindustryproj/service/dto/customs/EffectivenessPhaseModels/FinalEffectivenessPhaseReportModel.java
package com.marineindustryproj.service.dto.customs.EffectivenessPhaseModels;

import java.util.List;

public class FinalEffectivenessPhaseReportModel {

    private Integer levelId;

    private String levelTitle;

    private Long finishedCount;

    private Float averageEffectiveness;

    private List<DetailFinalEffectivenessPhaseReportModel> detailFinalEffectivenessPhaseReportModels;

    public FinalEffectivenessPhaseReportModel(Integer levelId, String levelTitle, Long finishedCount, Float averageEffectiveness, List<DetailFinalEffectivenessPhaseReportModel> detailFinalEffectivenessPhaseReportModels) {
        this.levelId = levelId;
        this.levelTitle = levelTitle;
        this.finishedCount = finishedCount;
        this.averageEffectiveness = averageEffectiveness;
        this.detailFinalEffectivenessPhaseReportModels = detailFinalEffectivenessPhaseReportModels;
    }

    public String getLevelTitle() {
        return levelTitle;
    }

    public void setLevelTitle(String levelTitle) {
        this.levelTitle = levelTitle;
    }

    public Long getFinishedCount() {
        return finishedCount;
    }

    public void setFinishedCount(Long finishedCount) {
        this.finishedCount = finishedCount;
    }

    public Float getAverageEffectiveness() {
        return averageEffectiveness;
    }

    public void setAverageEffectiveness(Float averageEffectiveness) {
        this.averageEffectiveness = averageEffectiveness;
    }

    public List<DetailFinalEffectivenessPhaseReportModel> getDetailFinalEffectivenessPhaseReportModels() {
        return detailFinalEffectivenessPhaseReportModels;
    }

    public void setDetailFinalEffectivenessPhaseReportModels(List<DetailFinalEffectivenessPhaseReportModel> detailFinalEffectivenessPhaseReportModels) {
        this.detailFinalEffectivenessPhaseReportModels = detailFinalEffectivenessPhaseReportModels;
    }

    public Integer getLevelId() {
        return levelId;
    }

    public void setLevelId(Integer levelId) {
        this.levelId = levelId;
    }
}
